### PR TITLE
wasm2js testing improvements before bulk-memory

### PR DESCRIPTION
**Pull request by kripken**
Necessary preparations for a later PR that adds bulk memory
support to wasm2js (splitting this out so the next PR is less big):

* Fix logging of `print` functions in wasm2js spec tests, there
was an extra space in the output (which `console.log` adds
automatically between items).

* Don't assume the output is always empty, as some tests
(like bulk memory) do have expected output.

* Rename `test/bulk-memory.wast` as it "conflicts" with
`test/spec/bulk-memory.wast` - the problem is that we scan
both places, and emit files in `test/wasm2js/*`, so those
would collide if the names overlap.

### CODE REVIEW

**Comment by kripken** [comment]
I thought I could split off #2919 from this, but it is actually necessary for tests to pass, so this also has

* Extend the parsing and emitting of JS for `(assert.. )` lines such as

  `(assert_return (invoke "foo" (i32.const 1)) (i32.const 2))`

  to also handle

  `(invoke "foo" (i32.const 1)) (i32.const 2))`

  Without this, we skip (invoke ..) lines in spec tests, which normally is
  fine, but in bulk memory at least they have side effects we need - we
  must run them before the later assertions.

File: src/tools/wasm2js.cpp [hunk 1]
```diff
@@ -690,6 +695,28 @@ Ref AssertionEmitter::emitAssertTrapFunc(Builder& wasmBuilder,
...
+                             body->type,
+                             std::vector<NameType>{},
+                             body));
+  Ref jsFunc = processFunction(testFunc.get());
+  fixCalls(jsFunc, asmModule);
+  emitFunction(jsFunc);
+  return jsFunc;
+}
+
+bool AssertionEmitter::isInvokeHandled(Element& e) {
```
Comment (tlively) [inline src/tools/wasm2js.cpp:715]: Why are these methods `isInvokeHandled` rather than just `isInvoke`? What is being handled?

**Comment by kripken** [comment]
The test renamings are the third bullet point in the first comment,
```
Rename test/bulk-memory.wast as it "conflicts" with
test/spec/bulk-memory.wast - the problem is that we scan
both places, and emit files in test/wasm2js/*, so those
would collide if the names overlap.
```

**Comment by tlively** [comment]
> The test renamings are the third bullet point in the first comment,

Oh my bad, I totally missed that. Does this also affect the other file names that are shared between `test/` and `test/spec`? The full list is atomics.wast, bulk-memory.wast, events.wast, exception-handling.wast, multivalue.wast, and  simd.wast.

**Comment by kripken** [comment]
It's not a problem with those yet, I guess - when the problem happens, it's a big error that's impossible to ignore, at least. But we may want to enable more of those eventually... so we should maybe solve this in a more general way, like adding a `spec/` prefix to spec outputs. That's a larger change though, so landing this for now.

**Pull request merged**